src/webgpu/util/texture/texelData.spec.ts

```diff
@@ -0,0 +1,377 @@
+      .expand(({ format }) => {
+        const max = (bitLength: number) => Math.pow(2, bitLength - 1) - 1;
+        return [
+          // Test extrema
+          makeParam(format, () => 0),
+          makeParam(format, bitLength => max(bitLength)),
+          makeParam(format, bitLength => -max(bitLength)),
+          makeParam(format, bitLength => -max(bitLength) - 1),
+
+          // Test a middle value
+          makeParam(format, bitLength => Math.floor(max(bitLength) / 2)),
+
+          // Test mixed values
+          makeParam(format, (bitLength, i) => {
+            const offset = [0.13, 0.63, 0.42, 0.89];
+            const range = 2 * max(bitLength);
+            return -max(bitLength) + Math.floor(offset[i] * range);
+          }),
+        ];
+      })
+  )
+  .fn(testFn);
+
+g.test('float_texel_data_in_shader')
+  .params(
+    params()
+      .combine(poptions('format', kEncodableTextureFormats))
+      .filter(({ format }) => {
+        return (
+          kEncodableTextureFormatInfo[format].copySrc &&
+          kEncodableTextureFormatInfo[format].copyDst &&
+          kEncodableTextureFormatInfo[format].color &&
+          kEncodableTextureFormatInfo[format].dataType === 'float'
+        );
+      })
+      .expand(({ format }) => {
+        return [
+          // Test extrema
+          makeParam(format, () => 0),
+
+          // TODO: Test NaN, Infinity, -Infinity
+
+          // Test some values
+          makeParam(format, () => 0.1199951171875),
+          makeParam(format, () => 1.4072265625),
+          makeParam(format, () => 24928),
+          makeParam(format, () => -0.1319580078125),
+          makeParam(format, () => -323.25),
+          makeParam(format, () => -7440),
+
+          // Test mixed values
+          makeParam(format, (bitLength, i) => {
+            const values = [24896, -0.1319580078125, -323.25, -234.375];
+            return values[i];
+          }),
+        ];
+      })
+  )
+  .fn(testFn);
+
+g.test('ufloat_texel_data_in_shader')
+  .params(
+    params()
+      .combine(poptions('format', kEncodableTextureFormats))
+      .filter(({ format }) => {
+        return (
+          kEncodableTextureFormatInfo[format].copySrc &&
+          kEncodableTextureFormatInfo[format].copyDst &&
+          kEncodableTextureFormatInfo[format].color &&
+          kEncodableTextureFormatInfo[format].dataType === 'ufloat'
+        );
+      })
+      .expand(({ format }) => {
+        return [
+          // Test extrema
+          makeParam(format, () => 0),
+
+          // TODO: Test NaN, Infinity
+
+          // Test some values
+          makeParam(format, () => 0.119873046875),
+          makeParam(format, () => 1.40625),
+          makeParam(format, () => 24896),
+
+          // Test mixed values
+          makeParam(format, (bitLength, i) => {
+            if (format === 'rgb9e5ufloat') {
+              // These values can't be too scattered or the representation
+              // will be very imprecise.
```

Comment:
We should also test values that are scattered though, as long as we know what to expect from the test (and I think we should be able to here)